### PR TITLE
DEV029 Create React_component for Musician_Skill model

### PR DESCRIPTION
**Pull request by YanPam**
Multi Select
[card](https://trello.com/c/CFpPhCTQ/73-dev029-create-reactcomponent-for-musicianskill-model)
Screenshot from 2019-08-22 11-00-46

### CODE REVIEW

File: app/javascript/components/MusicianSkillSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class MusicianSkillSelect extends React.Component {
+  state = {
+    selectedOption: null,
+  };
+  handleChange = selectedOption => {
+    this.setState({ selectedOption }); 
+  };
+  render() {
+    const { selectedOption } = this.state;
+    let options = []
```
[the remark this answers is not in the document]
Comment (YanPam) [reply app/javascript/components/MusicianSkillSelect.jsx]: This array is needed, without it does not work

[the remark this answers is not in the document]
Comment (YanPam) [reply app/javascript/components/MusicianSkillSelect.jsx]: Sorry, I was wrong.

File: app/javascript/components/MusicianSkillSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Select from 'react-select';
+
+class MusicianSkillSelect extends React.Component {
+  constructor(props) {
```
Comment (andreydro) [inline app/javascript/components/MusicianSkillSelect.jsx]: get rid of constructor

File: app/javascript/components/MusicianSkillSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    super(props);
+    this.state = { selectedOption: null };
+  }
+  handleChange = selectedOption => {
+    this.setState({ selectedOption }); 
+  };
+  options = this.props.musician_skills_array.map((skill) =>
```
Comment (andreydro) [inline app/javascript/components/MusicianSkillSelect.jsx]: destruct props

File: app/javascript/components/MusicianSkillSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    ( { value: skill.id, label: skill.name } )); 
+  render() {
+    const { selectedOption } = this.state;
```
Comment (andreydro) [inline app/javascript/components/MusicianSkillSelect.jsx]: fix indentation

File: yarn.lock [hunk 5]
```diff
@@ -693,7 +693,11 @@
     "@babel/plugin-transform-react-jsx-self" "^7.0.0"
     "@babel/plugin-transform-react-jsx-source" "^7.0.0"
 
+<<<<<<< HEAD
```
Comment (andreydro) [inline yarn.lock]: If you start to resolve conflict, then finish it

File: yarn.lock [hunk 6]
```diff
@@ -738,15 +742,89 @@
...
+    "@emotion/serialize" "^0.11.8"
+    "@emotion/utils" "0.11.2"
+    babel-plugin-emotion "^10.0.14"
+
+"@emotion/hash@0.7.2", "@emotion/hash@^0.7.1":
   version "0.7.2"
   resolved "https://registry.yarnpkg.com/@emotion/hash/-/hash-0.7.2.tgz#53211e564604beb9befa7a4400ebf8147473eeef"
   integrity sha512-RMtr1i6E8MXaBWwhXL3yeOU8JXRnz8GNxHvaUfVvwxokvayUY0zoBeWbKw1S9XkufmGEEdQd228pSZXFkAln8Q==
 
+<<<<<<< HEAD
```
Comment (andreydro) [inline yarn.lock]: If you start to resolve conflict, then finish it

File: yarn.lock [hunk 7]
```diff
@@ -5919,6 +6028,7 @@ promise-inflight@^1.0.1:
   resolved "https://registry.yarnpkg.com/promise-inflight/-/promise-inflight-1.0.1.tgz#98472870bf228132fcbdd868129bad12c3c029e3"
   integrity sha1-mEcocL8igTL8vdhoEputEsPAKeM=
 
+<<<<<<< HEAD
```
Comment (andreydro) [inline yarn.lock]: If you start to resolve conflict, then finish it

File: yarn.lock [hunk 8]
```diff
@@ -6109,7 +6229,18 @@ react-dom@^16.9.0:
     prop-types "^15.6.2"
     scheduler "^0.15.0"
 
+<<<<<<< HEAD
```
Comment (andreydro) [inline yarn.lock]: If you start to resolve conflict, then finish it

File: yarn.lock [hunk 9]
```diff
@@ -6119,6 +6250,7 @@ react-is@^16.8.1:
   resolved "https://registry.yarnpkg.com/react-is/-/react-is-16.8.6.tgz#5bbc1e2d29141c9fbdfed456343fe2bc430a6a16"
   integrity sha512-aUk3bHfZ2bRSVFFbbeVS4i+lNPZr3/WM5jT2J5omUVV1zzcs1nAaf3l51ctA5FFvCRbhrH0bdAsRRQddFJZPtA==
 
+<<<<<<< HEAD
```
Comment (andreydro) [inline yarn.lock]: If you start to resolve conflict, then finish it

File: yarn.lock [hunk 10]
```diff
@@ -6183,6 +6305,27 @@ react-search-field@^1.0.0:
...
+    "@emotion/cache" "^10.0.9"
+    "@emotion/core" "^10.0.9"
+    "@emotion/css" "^10.0.9"
+    classnames "^2.2.5"
+    memoize-one "^5.0.0"
+    prop-types "^15.6.0"
+    raf "^3.4.0"
+    react-input-autosize "^2.2.1"
+    react-transition-group "^2.2.1"
+>>>>>>> Create Musician Skill Multi Select
```
Comment (andreydro) [inline yarn.lock]: that is not suppose to be here

File: yarn.lock [hunk 11]
```diff
@@ -747,6 +782,47 @@
...
+"@emotion/utils@0.11.2":
+  version "0.11.2"
+  resolved "https://registry.yarnpkg.com/@emotion/utils/-/utils-0.11.2.tgz#713056bfdffb396b0a14f1c8f18e7b4d0d200183"
+  integrity sha512-UHX2XklLl3sIaP6oiMmlVzT0J+2ATTVpf0dHQVyPJHTkOITvXfaSqnRk6mdDhV9pR8T/tHc3cex78IKXssmzrA==
+
+"@emotion/weak-memoize@0.2.3":
+  version "0.2.3"
+  resolved "https://registry.yarnpkg.com/@emotion/weak-memoize/-/weak-memoize-0.2.3.tgz#dfa0c92efe44a1d1a7974fb49ffeb40ef2da5a27"
+  integrity sha512-zVgvPwGK7c1aVdUVc9Qv7SqepOGRDrqCw7KZPSZziWGxSlbII3gmvGLPzLX4d0n0BMbamBacUrN22zOMyFFEkQ==
+>>>>>>> Create Musician Skill Multi Select
```
Comment (andreydro) [inline yarn.lock]: that is not suppose to be here

**Pull request merged**